src/src/com/microsoft/aad/adal/AuthenticationContext.java

```diff
@@ -1453,17 +1454,22 @@ private AuthenticationResult localFlow(CallbackHandler callbackHandle,
                         new InstrumentationPropertiesBuilder(request, authResult).build());
                 return authResult;
             }
-        } 
+        }
         
         // refresh token does not exist or refresh token request failed to give back the token. 
         // If it's non-silent request, will try to acquire token interactively. 
         // if it's silent request, will return the AUTH_REFRESH_FAILED_PROMPT_NOT_ALLOWED.
         if (refreshItem == null || authResult == null 
                 || (authResult != null && StringExtensions.IsNullOrBlank(authResult.getAccessToken()))) {
             Logger.v(TAG, "Refresh token is not available or refresh token request failed to return token.");
+            Map<String, String> eventProperties = new InstrumentationPropertiesBuilder(request, authResult)
+                    .add(InstrumentationIDs.ERROR_CLASS, refreshItem == null ?
+                            InstrumentationIDs.REFRESH_TOKEN_NOT_FOUND :
+                            authResult == null ? InstrumentationIDs.AUTH_RESULT_EMPTY : InstrumentationIDs.AUTH_RESULT_CONTAINS_ERROR)
```

Comment:
AUTH_RESULT_CONTAINS_ERROR cannot be assumed. In this case, authResult does not contain an access token.